src/amo/components/Collection/index.js

```diff
@@ -247,15 +267,38 @@ export class CollectionBase extends React.Component<InternalProps> {
     invariant(slug, 'slug is required');
     invariant(username, 'username is required');
 
+    const isNewPage = switchCollectionPage(collection);
+
+    let page = filters.page;
+
+    if (isNewPage) {
+      page = Math.ceil(getCollectionPage(collection)) || 1;
```

Comment:
Why do you need `Math.ceil` here? That should be done in `getCollectionPage`, no?

Reply:
I was going to move it up there but then swithcCollectionPage would always return true here so I decided to leave this way. I guess I can make this into another function if that would make it more clear..?